packages/core/src/rules/oas3/component-name-unique.ts

```diff
@@ -46,26 +49,27 @@ export const ComponentNameUnique: Oas3Rule | Oas2Rule = (options) => {
           const resolvedRef = resolve(ref);
           if (!resolvedRef.location) return;
 
-          addComponentFromAbsoluteLocation(
-            typeName,
-            resolvedRef.location.absolutePointer.toString()
-          );
+          addComponentFromAbsoluteLocation(typeName, resolvedRef.location);
         }
       },
     },
     Root: {
       leave(root: Oas3Definition, ctx: UserContext) {
         components.forEach((value, key, _) => {
-          if (value.size > 1) {
+          if (value.absolutePointers.size > 1) {
             const component = getComponentFromKey(key);
             const optionComponentName = getOptionComponentNameForTypeName(component.typeName);
-            const definitions = Array.from(value)
+            const definitions = Array.from(value.absolutePointers)
               .map((v) => `- ${v}`)
               .join('\n');
-
             const problem: Problem = {
               message: `Component '${optionComponentName}/${component.componentName}' is not unique. It is defined at:\n${definitions}`,
             };
+            problem.location = {
+              ...value.locations.sort((a, b) =>
+                a.absolutePointer.localeCompare(b.absolutePointer)
+              )[0],
```

Comment:
Could we simply report each duplicated component? To me, it seems that would make spotting duplication easier.

Reply:
>  But I guess having only one on output but multiple in ignore is not possible, or is it?

I don't think it's possible. You can only ignore items that are in the output.

> Then those two reports basically are duplicates just with a different location

Perhaps you could filter out the current location and list only the duplicates to reduce noise. In most cases, there will likely be only one duplicate.

Reply:
Frankly, I'm not against having a single output location. I'm just curious if there’s a better option. If your approach works for you, let’s go with it 👍

Reply:
Change pushed.